doc/cost-model-overview/cost-model-overview.tex

```diff
@@ -0,0 +1,418 @@
+and ensure that the evaluation costs of scripts already on the chain remain
+unchanged, which is important when the history of the chain is replayed.
+
+A specific set of values for the machine step costs and the coefficients of the
+costing functions is referred to as a set of \textit{cost model parameters}
+(although \textit{cost model coefficients} might be a better name).  The Cardano
+ledger protocol parameters contain cost model parameters for each Plutus
+language version and each protocol version, and these are supplied to the Plutus
+Core evaluator before a script is evaluated to ensure that the execution costs
+are calculated appropriately.
+
+\subsection{Measuring costs during script execution}  Plutus Core scripts
+are deterministic: it is always known in advance exactly what the arguments of a
+script will be when it is executed on the chain, and hence it is also known
+exactly how the execution will proceed.  This means that an overall cost can be
+assigned to a program by running it and adding up the total costs of the machine
+steps and the builtin calls.  This is done by running the Plutus Core evaluator
+in \textit{counting mode}.  When a program is run on the chain the submitter
+supplies an expected CPU and memory budget (which they pay for in Ada).  To
+prevent denial of service it is necessary to check that the submitter has not
+lied about the budget, and this is done by running the evaluator in
+\textit{restricting mode}: it is supplied with the claimed budget and this is
+decremented as evaluation proceeds, with an error occurring if either the CPU or
+memory component ever become negative.  Keeping track of costs adds some
+overhead to execution times, and to minismise this we allow some
+\textit{slippage} in machine step costs.  A charge is made for the execution of
+each machine step, but the on-chain evaluator only checks that the total budget
+has not been exceeded once every 200 machine steps: this allows a program to
+perform slightly more computation than has been budgeted for but reduces the
+costing overhead significantly.  Slippage only applies to machine steps: the
+cost of evaluating a built-in function is calculated just before it is called
+and it is checked that calling it willnot cause the script's budget to be
+exceeded, so we never call a function if its execution will be too expensive.
+
+%% This is what necessitates the whole apparatus of costing functions: we can't
+%% instrument all of our builtins to calculate costs as they proceed, and we can't
+%% monitor execution dynamically: once we hand over control to a library function
+%% we just have to wait until it returns, so we have to be confident in advance
+%% that it is safe to call it.
+
+There are per-script limits for CPU and memory usage of scripts on the chain.
+These are the \texttt{steps} and \texttt{memory} components of the
+\texttt{maxTxExecution\-Units} protocol parameter and in Protocol Version 9 they
```

Comment:
Say something like "As of this writing they have values of ...", since their values may change without changing the PV.

Reply:
Wait, is that right?  I thought it needed a protocol update to change the protocol parameters.